IBM/CAST
# csmdb: schema migration 18.0 updated related to csm_switch tables and…

… specific key

# Purpose
Modified some dependencies related to the csm_switch and csm_switch_inventory table.  These changes will assist with the API and csm switch standardizing mellanox guids #697.

# Screenshots
```
csm_switch (table)
Indexes:
    "csm_switch_pkey" PRIMARY KEY, btree (switch_name)
    "uk_csm_switch_gu_id_a" UNIQUE CONSTRAINT, btree (gu_id)
Referenced by:
    TABLE "csm_switch_inventory" CONSTRAINT "csm_switch_inventory_host_system_guid_fkey" FOREIGN KEY (host_system_guid) REFERENCES csm_switch(gu_id)
--------------------------------------------------------------------------------------------------------
csm_switch_inventory (table)
Indexes:
    "csm_switch_inventory_pkey" PRIMARY KEY, btree (name)
Foreign-key constraints:
    "csm_switch_inventory_host_system_guid_fkey" FOREIGN KEY (host_system_guid) REFERENCES csm_switch(gu_id)
```

# How to Test
1. Run regression to ensure these updates don't break anything.
2. Tests with pre-populated data in the csm_switch and csm_switch inventory tables (before db schema migration 18.0 process). Then run the migration script so see if all upgrades have taken place.

## Open Questions and Pre-Merge TODOs
_Make sure you attempted to do the following:_
- [x] @pdlun92 to review (regression/testing)
- [x] @NickyDaB
